### PR TITLE
Fix duplicate detection, simpler regex, support custom ClassName

### PR DESCRIPTION
**Pull request by chillu**

### CODE REVIEW

**Comment by chillu** [comment]
@adiwidjaja Keen to peer review this? There's not a whole lot of maintainers or committers on this module. If nobody responds with any objects within a week I'm going to merge this myself

**Comment by adiwidjaja** [comment]
@chillu I will take a look later today. Nice to hear from you!

**Comment by adiwidjaja** [comment]
Ok, I tested it and everything looks fine! But the Tests don't run using /dev/tests/SiteTreeImporterTest, am I doing something wrong (other tests run)?

**Comment by chillu** [comment]
Hey Andy, great to hear from you as well! :) That's the right way to run the tests, are you doing it from the commandline or the browser? You might need to flush, keep in mind these two modes use separate caches. CLI flush with `sake dev/tests/SiteTreeImporterTest flush=all`.

**Comment by adiwidjaja** [comment]
Thanks for the tip, but I'm still getting:

    F
    SiteTreeImporterTest::testImport
    Failed asserting that null is an instance of class "Page".
    PHPUnit_Framework_Assert::assertInstanceOf(Page,)
    SiteTreeImporterTest.php:35

And four more errors. But if it runs for you, it's surely a problem of my setup.

File: README.md [hunk 1]
```diff
@@ -68,20 +68,27 @@ a key named `LegacyURL` is found in the imported JSON data.
...
-	public function setLegacyURL($legacyUrl) {
-		if(!$this->URLSegment) {
-			$this->URLSegment = $this->generateURLSegment();
-			$this->write();
+	public function setLegacyURL($url) {
+		$url = Director::makeRelative($url);
+		$urlBase = parse_url($url, PHP_URL_PATH);
+		$urlQuerystring = parse_url($url, PHP_URL_QUERY);
+
+		$urlObj = RedirectedURL::get()->filter(array(
```
Comment (madmatt) [inline README.md]: This keeps a dependency on the redirectedurls module, but that isn't specified in the composer.json - it should be added as a required module, right?

Comment (chillu) [inline README.md]: Nah, it mentions the module in the README above the code example, so its assumed that you install the module if you want that specific howto working.

**Comment by chillu** [comment]
Thanks for the feedback guys, merging.

**Pull request merged**